commands/update-ui/update-ui.sh

```diff
@@ -11,20 +11,31 @@ main() {
   cmd_runner_setup
 
   get_arg required --rust_version "$@"
-  rust_version="${out:-""}"
+  RUST_VERSION="${out:-""}"
 
   get_arg required --target_path "$@"
   target_path="${out:-""}"
 
-  if [[ -z "${rust_version}" ]]; then
-    die "missing rust version argument"
+
+  if [[ ! -z "${RUST_VERSION}" ]]; then
+    rustup install $RUST_VERSION
+    rustup component add rust-src --toolchain $RUST_VERSION
   fi
 
-  # This script uses rustup to install the required rust version.
-  # Doing that in CI feels rather ugly, but sadly there's currently no mechanism
-  # in command-bot to use a different CI image for individual jobs, so this is
-  # the best we can do.
-  "$target_path/.maintain/update-rust-stable.sh" "${rust_version}"
+  # Ensure we run the ui tests
+  export RUN_UI_TESTS=1
+  # We don't need any wasm files for ui tests
+  export SKIP_WASM_BUILD=1
+  # Let trybuild overwrite the .stderr files
+  export TRYBUILD=overwrite
+
+  # Run all the relevant UI tests
+  #
+  # Any new UI tests in different crates need to be added here as well.
+  rustup run $RUST_VERSION cargo test -p sp-runtime-interface ui
+  rustup run $RUST_VERSION cargo test -p sp-api-test ui
+  rustup run $RUST_VERSION cargo test -p frame-election-provider-solution-type ui
+  rustup run $RUST_VERSION cargo test -p frame-support-test ui
```

Comment:
```suggestion
  # ${RUST_VERSION:++$RUST_VERSION} expands to +$RUST_VERSION if it is set, "" otherwise
  cargo ${RUST_VERSION:++$RUST_VERSION} test -p sp-runtime-interface ui
  cargo ${RUST_VERSION:++$RUST_VERSION} test -p sp-api-test ui
  cargo ${RUST_VERSION:++$RUST_VERSION} test -p frame-election-provider-solution-type ui
  cargo ${RUST_VERSION:++$RUST_VERSION} test -p frame-support-test ui
```
You've made `$RUST_VERSION` an optional argument, and `rustup run "" cargo ...` isn't gonna work.